scripts/cluster/agent.py

```diff
@@ -423,6 +443,285 @@ def configure():
+        error_msg = {"error": "Empty service provided"}
+        return Response(json.dumps(error_msg), mimetype='application/json', status=400)
+    output = subprocess.check_output("systemctl restart snap.{}.service".format(request.json["service"]).split())
+    return output
+
+
+@app.route('/{}/service/start'.format(CLUSTER_API), methods=['POST'])
+def service_start():
+    """
+    Web call to start a service
+    :request_param service: the name of the service
+    """
+    v = callback_token_validation(request)
+    if not v["valid"]: return v["response"]
+    if "service" not in request.json or len(request.json["service"].strip()) == 0:
+        error_msg = {"error": "Empty service provided"}
+        return Response(json.dumps(error_msg), mimetype='application/json', status=400)
+    output = subprocess.check_output("systemctl start snap.{}.service".format(request.json["service"]).split())
+    return output
+
+
+@app.route('/{}/service/stop'.format(CLUSTER_API), methods=['POST'])
+def service_stop():
+    """
+    Web call to start a service
+    :request_param service: the name of the service
+    """
+    v = callback_token_validation(request)
+    if not v["valid"]: return v["response"]
+    if "service" not in request.json or len(request.json["service"].strip()) == 0:
+        error_msg = {"error": "Empty service provided"}
+        return Response(json.dumps(error_msg), mimetype='application/json', status=400)
+    output = subprocess.check_output("systemctl stop snap.{}.service".format(request.json["service"]).split())
+    return output
+
+
+@app.route('/{}/service/enable'.format(CLUSTER_API), methods=['POST'])
+def service_enable():
+    """
+    Web call to enable a service
+    :request_param service: the name of the service
+    """
+    v = callback_token_validation(request)
+    if not v["valid"]: return v["response"]
+    if "service" not in request.json or len(request.json["service"].strip()) == 0:
+        error_msg = {"error": "Empty service provided"}
+        return Response(json.dumps(error_msg), mimetype='application/json', status=400)
+    output = subprocess.check_output("systemctl enable snap.{}.service".format(request.json["service"]).split())
+    return output
+
+
+@app.route('/{}/service/disable'.format(CLUSTER_API), methods=['POST'])
+def service_disable():
+    """
+    Web call to enable a service
+    :request_param service: the name of the service
+    """
+    v = callback_token_validation(request)
+    if not v["valid"]: return v["response"]
+    if "service" not in request.json or len(request.json["service"].strip()) == 0:
+        error_msg = {"error": "Empty service provided"}
+        return Response(json.dumps(error_msg), mimetype='application/json', status=400)
+    output = subprocess.check_output("systemctl disable snap.{}.service".format(request.json["service"]).split())
+    return output
+
+
+@app.route('/{}/service/logs'.format(CLUSTER_API), methods=['POST'])
```

Comment:
Could you put the log collection under the status endpoint? Thank you.

Reply:
Do you mean /status/service/logs? Or extend the json body which is sent to /status, to support the service logs with new properties?

eg on /status:
`{"callback":"xyztoken", "servicelogs":{"name":"microk8s.daemon-flanneld", "lines":30}}